serverless/src/deleteRetrieval/handler.js

```diff
@@ -35,6 +35,11 @@ export default async function deleteRetrieval(event, context) {
     // Retrieve a connection to the database
     const dbConnection = await getDbConnection()
 
+    // Nullify the foreign key constraint on the table before deleting
+    await dbConnection('retrieval_collections')
+      .where('retrieval_id', '=', decodedRetrievalId)
```

Comment:
For future self, `.where('retrieval_id', decodedRetrievalId)` and is a bit less typing, I would typically only use `=` if the = was something that wasn't the default, like `>` or something.